src/core/declarativewebcontainer.cpp

```diff
@@ -864,11 +873,18 @@ void DeclarativeWebContainer::initialize()
     // 2) model has tabs, load initial url or active tab.
     bool firstUseDone = DeclarativeWebUtils::instance()->firstUseDone();
     if ((m_model->waitingForNewTab() || m_model->count() == 0) && (firstUseDone || !m_initialUrl.isEmpty())) {
-        QString url = m_initialUrl.isEmpty() ? DeclarativeWebUtils::instance()->homePage() : m_initialUrl;
+        QString url = m_initialUrl;
+        if (m_initialUrl.isEmpty()) {
+            if (!browserEnabled()) url = ABOUT_BLANK;
```

Comment:
```
if (!browserEnabled())
    url = ABOUT_BLANK;
else
    url = DeclarativeWebUtils::instance()->homePage();
```

Reply:
Done. Though I think this should happen in all privateMode, rather than just when the browser is disabled.